### PR TITLE
[KMS]: update envelope caching unit tests

### PR DESCRIPTION
**Pull request by aramase**
Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
- Updates `TestEnvelopeCaching` unit test to simulate failures when cache is disabled

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/envelope_test.go [hunk 1]
```diff
@@ -83,15 +83,23 @@ func TestEnvelopeCaching(t *testing.T) {
 		desc                     string
 		cacheSize                int
 		simulateKMSPluginFailure bool
+		expectedError            bool
```
Comment (enj) [inline staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/envelope_test.go:86]: Can we check that the error (or a substring) actually matches what we want (same comment for the other file)?

```suggestion
		expectedError            string
```

Comment (aramase) [reply staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/envelope_test.go]: updated!

**Pull request merged**